clerk-frontend/src/need-wizard/ConfirmStep.js

```diff
@@ -44,37 +46,42 @@ class ConfirmStep extends Component {
       'data': result.claim.data
     })
   }
-
-  renderAttributes() {
-    let result = []
-    if (this.state.data) {
-      for (let keyValue of this.state.data) {
-        let key = Object.keys(keyValue)[0]
-        let value = keyValue[key]
-
-        if (typeof value !==  'string') {
-          value = JSON.stringify(value)
+  
+  renderAttributes(obj, recursive) {
+    if (!recursive) {
+      // Clear existing array - this is required when user selects 'vorige stap' in Clerk Frontend
+      result = []
+    }
+    if (obj) {
+      for (var k in obj) {
```

Comment:
It's probably a little safer to iterate with (const k in Object.keys(obj)) instead, it prevents accidentally iterating over keys that are inherited.

Reply:
This is weird. While working on the recursive rendering, I couldn't figure out how to get rid of the keys apparently linked to the SOURCE_DATA_COLLECTION fields(?), and choose to leave it like that. With this suggestion however I only get those keys back?! I'm confused...

`for (const k in Object.keys(obj) {` shows:

![Screenshot from 2020-04-03 09-35-22](https://user-images.githubusercontent.com/25812095/78335907-cb030500-758e-11ea-876c-2dac0085a3fd.png)


And `for (var k in obj) {` shows:

![Screenshot from 2020-04-03 09-45-03](https://user-images.githubusercontent.com/25812095/78336556-d73b9200-758f-11ea-8b7f-5df71748a25d.png)

Reply:
Okay, so lets dive into this a little deeper.

It turns out we're using that particular loop both for arrays and objects. 

I think we actually want to separate in those cases, since in the case of the object we likely do want to print the key, whereas in the case of arrays the key is just a number and we don't want to print it.

Reply:
Interesting. And nice learning experience. :+1: 
If I use template literals in the print statement of `$obj[k]}`, Javascript shows _undefined_. If I don't, Javascript shows nothing. As expected indeed. But hard to recognize while debugging...

![Screenshot from 2020-04-03 14-54-21](https://user-images.githubusercontent.com/25812095/78362832-0f0bff00-75bb-11ea-9da5-ea4bb8431a99.png)

Reply:
Fixed with the help of te master himself. Thank you for that!

Reply:
Will figure out how to improve current presentation. Wait for that commit happening and then consider this PR ready for merge.

Reply:
Consider this done.

![Screenshot from 2020-04-03 18-33-12](https://user-images.githubusercontent.com/25812095/78384011-b26c0c80-75d9-11ea-9d51-f0e8e54682bb.png)